Source/WebKit/WebProcess/Plugins/PDF/PDFPluginBase.mm

```diff
@@ -910,6 +1112,48 @@
     return isFullFramePlugin();
 }
 
+// Accessibility Related
+bool PDFPluginBase::showContextMenuAtPoint(const IntPoint& point)
+{
+    auto* frameView = m_frame ? m_frame->coreLocalFrame()->view() : nullptr;
+    if (!frameView)
+        return false;
+    IntPoint contentsPoint = frameView->contentsToRootView(point);
+    WebMouseEvent event({ WebEventType::MouseDown, OptionSet<WebEventModifier> { }, WallTime::now() }, WebMouseEventButton::Right, 0, contentsPoint, contentsPoint, 0, 0, 0, 1, WebCore::ForceAtClick);
+    return handleContextMenuEvent(event);
+}
+
+IntPoint PDFPluginBase::convertFromPDFViewToRootView(const IntPoint& point) const
+{
+    IntPoint pointInPluginCoordinates(point.x(), size().height() - point.y());
+    return valueOrDefault(m_rootViewToPluginTransform.inverse()).mapPoint(pointInPluginCoordinates);
+}
+
+FloatRect PDFPluginBase::convertFromPDFViewToScreen(const FloatRect& rect) const
```

Comment:
It's really weird for this utility function to call `Accessibility::retrieveValueFromMainThread`. In general, we should not call any functions on PDFPluginBase off the main thread; we don't consider this class to be thread safe at all.

Reply:
I copy this function from PDFPlugin.mm `FloatRect PDFPlugin::convertFromPDFViewToScreen(const FloatRect& rect) const`. Maybe I should not do that?

Reply:
It would be OK to rename it `convertFromPDFViewToScreenForAccessibility` for now and leave it, but I think we need to fix this soon.